<!DOCTYPE html>
<html lang="en-US" dir="ltr">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>PanVA: Homology app data format | PanVA</title>
    <meta name="description" content="Documentation for PanVA">
    <meta name="generator" content="VitePress v1.1.3">
    <link rel="preload stylesheet" href="/PanVA/assets/style.4xkFXuh2.css" as="style">
    
    <script type="module" src="/PanVA/assets/app.BZ7zcMaX.js"></script>
    <link rel="preload" href="/PanVA/assets/inter-roman-latin.Di8DUHzh.woff2" as="font" type="font/woff2" crossorigin="">
    <link rel="modulepreload" href="/PanVA/assets/chunks/framework.DonzCgw1.js">
    <link rel="modulepreload" href="/PanVA/assets/chunks/theme.BRIo-sX-.js">
    <link rel="modulepreload" href="/PanVA/assets/v0.0.0_data-format-homology.md.D4-vdtjc.lean.js">
    <script id="check-dark-mode">(()=>{const e=localStorage.getItem("vitepress-theme-appearance")||"auto",a=window.matchMedia("(prefers-color-scheme: dark)").matches;(!e||e==="auto"?a:e==="dark")&&document.documentElement.classList.add("dark")})();</script>
    <script id="check-mac-os">document.documentElement.classList.toggle("mac",/Mac|iPhone|iPod|iPad/i.test(navigator.platform));</script>
  </head>
  <body>
    <div id="app"><div class="Layout" data-v-014bb8c7><!--[--><!--]--><!--[--><span tabindex="-1" data-v-e8ee1042></span><a href="#VPContent" class="VPSkipLink visually-hidden" data-v-e8ee1042> Skip to content </a><!--]--><!----><header class="VPNav" data-v-014bb8c7 data-v-dd5cae27><div class="VPNavBar has-sidebar top" data-v-dd5cae27 data-v-184485b8><div class="wrapper" data-v-184485b8><div class="container" data-v-184485b8><div class="title" data-v-184485b8><div class="VPNavBarTitle has-sidebar" data-v-184485b8 data-v-61610785><a class="title" href="/PanVA/" data-v-61610785><!--[--><!--]--><!----><span data-v-61610785>PanVA</span><!--[--><!--]--></a></div></div><div class="content" data-v-184485b8><div class="content-body" data-v-184485b8><!--[--><!--]--><div class="VPNavBarSearch search" data-v-184485b8><!--[--><!----><div id="local-search"><button type="button" class="DocSearch DocSearch-Button" aria-label="Search"><span class="DocSearch-Button-Container"><span class="vp-icon DocSearch-Search-Icon"></span><span class="DocSearch-Button-Placeholder">Search</span></span><span class="DocSearch-Button-Keys"><kbd class="DocSearch-Button-Key"></kbd><kbd class="DocSearch-Button-Key">K</kbd></span></button></div><!--]--></div><nav aria-labelledby="main-nav-aria-label" class="VPNavBarMenu menu" data-v-184485b8 data-v-15f677bc><span id="main-nav-aria-label" class="visually-hidden" data-v-15f677bc>Main Navigation</span><!--[--><!--[--><a class="VPLink link VPNavBarMenuLink" href="/PanVA/" tabindex="0" data-v-15f677bc data-v-0a5152c9><!--[--><span data-v-0a5152c9>Home</span><!--]--></a><!--]--><!--[--><a class="VPLink link VPNavBarMenuLink" href="/PanVA/v0.0.0/install.html" tabindex="0" data-v-15f677bc data-v-0a5152c9><!--[--><span data-v-0a5152c9>Installation</span><!--]--></a><!--]--><!--[--><a class="VPLink link VPNavBarMenuLink" href="/PanVA/demos.html" tabindex="0" data-v-15f677bc data-v-0a5152c9><!--[--><span data-v-0a5152c9>Demos</span><!--]--></a><!--]--><!--[--><a class="VPLink link VPNavBarMenuLink" href="/PanVA/v0.0.0/tutorials.html" tabindex="0" data-v-15f677bc data-v-0a5152c9><!--[--><span data-v-0a5152c9>Tutorials</span><!--]--></a><!--]--><!--[--><div class="VPFlyout VPNavBarMenuGroup" data-v-15f677bc data-v-05ee7a22><button type="button" class="button" aria-haspopup="true" aria-expanded="false" data-v-05ee7a22><span class="text" data-v-05ee7a22><!----><span data-v-05ee7a22>Versions</span><span class="vpi-chevron-down text-icon" data-v-05ee7a22></span></span></button><div class="menu" data-v-05ee7a22><div class="VPMenu" data-v-05ee7a22 data-v-b14807a1><div class="items" data-v-b14807a1><!--[--><!--[--><div class="VPMenuLink" data-v-b14807a1 data-v-51cec001><a class="VPLink link" href="/PanVA/v0.0.0/install.html" data-v-51cec001><!--[-->0.0.0 (latest)<!--]--></a></div><!--]--><!--]--></div><!--[--><!--]--></div></div></div><!--]--><!--]--></nav><!----><div class="VPNavBarAppearance appearance" data-v-184485b8 data-v-1eb84711><button class="VPSwitch VPSwitchAppearance" type="button" role="switch" title="Switch to dark theme" aria-checked="false" data-v-1eb84711 data-v-a9e811fe data-v-db85a47e><span class="check" data-v-db85a47e><span class="icon" data-v-db85a47e><!--[--><span class="vpi-sun sun" data-v-a9e811fe></span><span class="vpi-moon moon" data-v-a9e811fe></span><!--]--></span></span></button></div><div class="VPSocialLinks VPNavBarSocialLinks social-links" data-v-184485b8 data-v-82da3dba data-v-cbd38fa4><!--[--><a class="VPSocialLink no-icon" href="https://github.com/PanBrowse/PanVA" aria-label="github" target="_blank" rel="noopener" data-v-cbd38fa4 data-v-3c68021a><span class="vpi-social-github" /></a><!--]--></div><div class="VPFlyout VPNavBarExtra extra" data-v-184485b8 data-v-03da2ef4 data-v-05ee7a22><button type="button" class="button" aria-haspopup="true" aria-expanded="false" aria-label="extra navigation" data-v-05ee7a22><span class="vpi-more-horizontal icon" data-v-05ee7a22></span></button><div class="menu" data-v-05ee7a22><div class="VPMenu" data-v-05ee7a22 data-v-b14807a1><!----><!--[--><!--[--><!----><div class="group" data-v-03da2ef4><div class="item appearance" data-v-03da2ef4><p class="label" data-v-03da2ef4>Appearance</p><div class="appearance-action" data-v-03da2ef4><button class="VPSwitch VPSwitchAppearance" type="button" role="switch" title="Switch to dark theme" aria-checked="false" data-v-03da2ef4 data-v-a9e811fe data-v-db85a47e><span class="check" data-v-db85a47e><span class="icon" data-v-db85a47e><!--[--><span class="vpi-sun sun" data-v-a9e811fe></span><span class="vpi-moon moon" data-v-a9e811fe></span><!--]--></span></span></button></div></div></div><div class="group" data-v-03da2ef4><div class="item social-links" data-v-03da2ef4><div class="VPSocialLinks social-links-list" data-v-03da2ef4 data-v-cbd38fa4><!--[--><a class="VPSocialLink no-icon" href="https://github.com/PanBrowse/PanVA" aria-label="github" target="_blank" rel="noopener" data-v-cbd38fa4 data-v-3c68021a><span class="vpi-social-github" /></a><!--]--></div></div></div><!--]--><!--]--></div></div></div><!--[--><!--]--><button type="button" class="VPNavBarHamburger hamburger" aria-label="mobile navigation" aria-expanded="false" aria-controls="VPNavScreen" data-v-184485b8 data-v-be6e58cf><span class="container" data-v-be6e58cf><span class="top" data-v-be6e58cf></span><span class="middle" data-v-be6e58cf></span><span class="bottom" data-v-be6e58cf></span></span></button></div></div></div></div><div class="divider" data-v-184485b8><div class="divider-line" data-v-184485b8></div></div></div><!----></header><div class="VPLocalNav has-sidebar empty" data-v-014bb8c7 data-v-fbe6f17f><div class="container" data-v-fbe6f17f><button class="menu" aria-expanded="false" aria-controls="VPSidebarNav" data-v-fbe6f17f><span class="vpi-align-left menu-icon" data-v-fbe6f17f></span><span class="menu-text" data-v-fbe6f17f>Menu</span></button><div class="VPLocalNavOutlineDropdown" style="--vp-vh:0px;" data-v-fbe6f17f data-v-adc9fa8c><button data-v-adc9fa8c>Return to top</button><!----></div></div></div><aside class="VPSidebar" data-v-014bb8c7 data-v-a2aeecee><div class="curtain" data-v-a2aeecee></div><nav class="nav" id="VPSidebarNav" aria-labelledby="sidebar-aria-label" tabindex="-1" data-v-a2aeecee><span class="visually-hidden" id="sidebar-aria-label" data-v-a2aeecee> Sidebar Navigation </span><!--[--><!--]--><!--[--><div class="group" data-v-a2aeecee><section class="VPSidebarItem level-0 collapsible" data-v-a2aeecee data-v-fef27b75><div class="item" role="button" tabindex="0" data-v-fef27b75><div class="indicator" data-v-fef27b75></div><h2 class="text" data-v-fef27b75>Getting Started</h2><div class="caret" role="button" aria-label="toggle section" tabindex="0" data-v-fef27b75><span class="vpi-chevron-right caret-icon" data-v-fef27b75></span></div></div><div class="items" data-v-fef27b75><!--[--><div class="VPSidebarItem level-1 is-link" data-v-fef27b75 data-v-fef27b75><div class="item" data-v-fef27b75><div class="indicator" data-v-fef27b75></div><a class="VPLink link link" href="/PanVA/v0.0.0/install.html" data-v-fef27b75><!--[--><p class="text" data-v-fef27b75>Installation</p><!--]--></a><!----></div><!----></div><!--]--></div></section></div><div class="group" data-v-a2aeecee><section class="VPSidebarItem level-0 collapsible has-active" data-v-a2aeecee data-v-fef27b75><div class="item" role="button" tabindex="0" data-v-fef27b75><div class="indicator" data-v-fef27b75></div><h2 class="text" data-v-fef27b75>User Guide</h2><div class="caret" role="button" aria-label="toggle section" tabindex="0" data-v-fef27b75><span class="vpi-chevron-right caret-icon" data-v-fef27b75></span></div></div><div class="items" data-v-fef27b75><!--[--><section class="VPSidebarItem level-1 is-link has-active" data-v-fef27b75 data-v-fef27b75><div class="item" tabindex="0" data-v-fef27b75><div class="indicator" data-v-fef27b75></div><a class="VPLink link link" href="/PanVA/v0.0.0/data-format.html" data-v-fef27b75><!--[--><h3 class="text" data-v-fef27b75>Data Format Reference</h3><!--]--></a><!----></div><div class="items" data-v-fef27b75><!--[--><div class="VPSidebarItem level-2 is-link" data-v-fef27b75 data-v-fef27b75><div class="item" data-v-fef27b75><div class="indicator" data-v-fef27b75></div><a class="VPLink link link" href="/PanVA/v0.0.0/data-format-homology.html" data-v-fef27b75><!--[--><p class="text" data-v-fef27b75>Homology</p><!--]--></a><!----></div><!----></div><!--]--></div></section><section class="VPSidebarItem level-1 is-link" data-v-fef27b75 data-v-fef27b75><div class="item" tabindex="0" data-v-fef27b75><div class="indicator" data-v-fef27b75></div><a class="VPLink link link" href="/PanVA/v0.0.0/config.html" data-v-fef27b75><!--[--><h3 class="text" data-v-fef27b75>Configuration</h3><!--]--></a><!----></div><div class="items" data-v-fef27b75><!--[--><div class="VPSidebarItem level-2 is-link" data-v-fef27b75 data-v-fef27b75><div class="item" data-v-fef27b75><div class="indicator" data-v-fef27b75></div><a class="VPLink link link" href="/PanVA/v0.0.0/config.html#example-config" data-v-fef27b75><!--[--><p class="text" data-v-fef27b75>Example config</p><!--]--></a><!----></div><!----></div><!--]--></div></section><div class="VPSidebarItem level-1 is-link" data-v-fef27b75 data-v-fef27b75><div class="item" data-v-fef27b75><div class="indicator" data-v-fef27b75></div><a class="VPLink link link" href="/PanVA/v0.0.0/server-config.html" data-v-fef27b75><!--[--><p class="text" data-v-fef27b75>Server setup</p><!--]--></a><!----></div><!----></div><!--]--></div></section></div><div class="group" data-v-a2aeecee><section class="VPSidebarItem level-0 collapsible" data-v-a2aeecee data-v-fef27b75><div class="item" role="button" tabindex="0" data-v-fef27b75><div class="indicator" data-v-fef27b75></div><h2 class="text" data-v-fef27b75>Demos & Tutorials</h2><div class="caret" role="button" aria-label="toggle section" tabindex="0" data-v-fef27b75><span class="vpi-chevron-right caret-icon" data-v-fef27b75></span></div></div><div class="items" data-v-fef27b75><!--[--><div class="VPSidebarItem level-1 is-link" data-v-fef27b75 data-v-fef27b75><div class="item" data-v-fef27b75><div class="indicator" data-v-fef27b75></div><a class="VPLink link link" href="/PanVA/v0.0.0/yeast-tutorial.html" data-v-fef27b75><!--[--><p class="text" data-v-fef27b75>Yeast</p><!--]--></a><!----></div><!----></div><div class="VPSidebarItem level-1 is-link" data-v-fef27b75 data-v-fef27b75><div class="item" data-v-fef27b75><div class="indicator" data-v-fef27b75></div><a class="VPLink link link" href="/PanVA/v0.0.0/pecto-tutorial.html" data-v-fef27b75><!--[--><p class="text" data-v-fef27b75>Pectobacterium</p><!--]--></a><!----></div><!----></div><div class="VPSidebarItem level-1 is-link" data-v-fef27b75 data-v-fef27b75><div class="item" data-v-fef27b75><div class="indicator" data-v-fef27b75></div><a class="VPLink link link" href="/PanVA/v0.0.0/ara-tutorial.html" data-v-fef27b75><!--[--><p class="text" data-v-fef27b75>Arabidopsis</p><!--]--></a><!----></div><!----></div><div class="VPSidebarItem level-1 is-link" data-v-fef27b75 data-v-fef27b75><div class="item" data-v-fef27b75><div class="indicator" data-v-fef27b75></div><a class="VPLink link link" href="/PanVA/pantools.html" data-v-fef27b75><!--[--><p class="text" data-v-fef27b75>PanTools</p><!--]--></a><!----></div><!----></div><!--]--></div></section></div><div class="group" data-v-a2aeecee><section class="VPSidebarItem level-0" data-v-a2aeecee data-v-fef27b75><!----><div class="items" data-v-fef27b75><!--[--><div class="VPSidebarItem level-1 is-link" data-v-fef27b75 data-v-fef27b75><div class="item" data-v-fef27b75><div class="indicator" data-v-fef27b75></div><a class="VPLink link link" href="/PanVA/v0.0.0/dev-setup.html" data-v-fef27b75><!--[--><p class="text" data-v-fef27b75>Development Guide</p><!--]--></a><!----></div><!----></div><!--]--></div></section></div><!--]--><!--[--><!--]--></nav></aside><div class="VPContent has-sidebar" id="VPContent" data-v-014bb8c7 data-v-a5b8dade><div class="VPDoc has-sidebar has-aside" data-v-a5b8dade data-v-b474ad16><!--[--><!--]--><div class="container" data-v-b474ad16><div class="aside" data-v-b474ad16><div class="aside-curtain" data-v-b474ad16></div><div class="aside-container" data-v-b474ad16><div class="aside-content" data-v-b474ad16><div class="VPDocAside" data-v-b474ad16 data-v-cc2b2345><!--[--><!--]--><!--[--><!--]--><div class="VPDocAsideOutline" role="navigation" data-v-cc2b2345 data-v-b1754174><div class="content" data-v-b1754174><div class="outline-marker" data-v-b1754174></div><div class="outline-title" role="heading" aria-level="2" data-v-b1754174>On this page</div><nav aria-labelledby="doc-outline-aria-label" data-v-b1754174><span class="visually-hidden" id="doc-outline-aria-label" data-v-b1754174> Table of Contents for current page </span><ul class="VPDocOutlineItem root" data-v-b1754174 data-v-557df799><!--[--><!--]--></ul></nav></div></div><!--[--><!--]--><div class="spacer" data-v-cc2b2345></div><!--[--><!--]--><!----><!--[--><!--]--><!--[--><!--]--></div></div></div></div><div class="content" data-v-b474ad16><div class="content-container" data-v-b474ad16><!--[--><!--]--><main class="main" data-v-b474ad16><div style="position:relative;" class="vp-doc _PanVA_v0_0_0_data-format-homology" data-v-b474ad16><div><h1 id="panva-homology-app-data-format" tabindex="-1">PanVA: Homology app data format <span class="VPBadge info"><!--[-->v0.0.0<!--]--></span> <a class="header-anchor" href="#panva-homology-app-data-format" aria-label="Permalink to &quot;PanVA: Homology app data format &lt;Badge type=&quot;info&quot; text=&quot;v0.0.0&quot; /&gt;&quot;">​</a></h1><p>This file provides a specification for the data format as used by the PanVA Homology app.</p><p>We recommend to run PanTools (<a href="https://pantools.readthedocs.io/en/stable/" target="_blank" rel="noreferrer">https://pantools.readthedocs.io/en/stable/</a>) to construct a pangenome database and run functionality to obtain homology groups, sequence alignments, annotations and metadata. We support this format and are currently working on a pipeline to preprocess the PanTools data such that it is formatted correctly for PanVA. More information about this pipeline will be released soon. However, it may be possible to obtain the analysis data using other software if the output can be transformed to the format required for PanVA.</p><p>Below we describe the data directory structure and the file names, formats, and data features of the files inside those directories.</p><h2 id="data-directory-structure" tabindex="-1">Data directory structure <a class="header-anchor" href="#data-directory-structure" aria-label="Permalink to &quot;Data directory structure&quot;">​</a></h2><p>The app data directory contains files containing data regarding the full dataset.</p><p>Each homology group is a subdirectory of the app data directory and has a unique name, corresponding to the <code>homology_id</code> identifier in <code>homologies.json</code>, and contains the files belonging to that homology group.</p><div class="language- vp-adaptive-theme"><button title="Copy Code" class="copy"></button><span class="lang"></span><pre class="shiki shiki-themes github-light github-dark vp-code"><code><span class="line"><span>/</span></span>
<span class="line"><span>+-- [other apps]</span></span>
<span class="line"><span>+-- homology/</span></span>
<span class="line"><span>    |-- homologies.json</span></span>
<span class="line"><span>    |-- [tree files]</span></span>
<span class="line"><span>    |</span></span>
<span class="line"><span>    +-- &lt;homology_id&gt;/</span></span>
<span class="line"><span>        |-- alignments.csv</span></span>
<span class="line"><span>        |-- sequences.csv</span></span>
<span class="line"><span>        |-- variable.csv</span></span>
<span class="line"><span>        |-- annotations.csv     (optional)</span></span>
<span class="line"><span>        |-- metadata.csv        (optional)</span></span>
<span class="line"><span>        |-- linkage_matrix.npy  (auto-generated)</span></span></code></pre></div><h2 id="dataset-files" tabindex="-1">Dataset files <a class="header-anchor" href="#dataset-files" aria-label="Permalink to &quot;Dataset files&quot;">​</a></h2><h3 id="homologies-json" tabindex="-1"><code>homologies.json</code> <a class="header-anchor" href="#homologies-json" aria-label="Permalink to &quot;`homologies.json`&quot;">​</a></h3><p>A list of objects representing all homology ids of the selected set. Homology id objects have the following properties:</p><ul><li><code>id</code>: Unique id for the homology group (<em>string</em>).</li><li><code>members</code>: Number of sequences (<em>integer</em>).</li><li><code>alignment_length</code>: Length of the alignment (<em>integer</em>).</li><li><code>metadata</code>: Metadata shown in the frontend (<em>object</em>). <ul><li><code>key</code>: Internal name (<em>string</em>).</li><li><code>value</code>: Value to display (<em>string, boolean, number, Array of strings</em>).</li></ul></li></ul><p>An example object in the array:</p><div class="language-json vp-adaptive-theme"><button title="Copy Code" class="copy"></button><span class="lang">json</span><pre class="shiki shiki-themes github-light github-dark vp-code"><code><span class="line"><span style="--shiki-light:#24292E;--shiki-dark:#E1E4E8;">[</span></span>
<span class="line"><span style="--shiki-light:#24292E;--shiki-dark:#E1E4E8;">    {</span></span>
<span class="line"><span style="--shiki-light:#005CC5;--shiki-dark:#79B8FF;">        &quot;id&quot;</span><span style="--shiki-light:#24292E;--shiki-dark:#E1E4E8;">: </span><span style="--shiki-light:#032F62;--shiki-dark:#9ECBFF;">&quot;13773385&quot;</span><span style="--shiki-light:#24292E;--shiki-dark:#E1E4E8;">,</span></span>
<span class="line"><span style="--shiki-light:#005CC5;--shiki-dark:#79B8FF;">        &quot;members&quot;</span><span style="--shiki-light:#24292E;--shiki-dark:#E1E4E8;">: </span><span style="--shiki-light:#005CC5;--shiki-dark:#79B8FF;">197</span><span style="--shiki-light:#24292E;--shiki-dark:#E1E4E8;">,</span></span>
<span class="line"><span style="--shiki-light:#005CC5;--shiki-dark:#79B8FF;">        &quot;alignment_length&quot;</span><span style="--shiki-light:#24292E;--shiki-dark:#E1E4E8;">: </span><span style="--shiki-light:#005CC5;--shiki-dark:#79B8FF;">996</span><span style="--shiki-light:#24292E;--shiki-dark:#E1E4E8;">,</span></span>
<span class="line"><span style="--shiki-light:#005CC5;--shiki-dark:#79B8FF;">        &quot;metadata&quot;</span><span style="--shiki-light:#24292E;--shiki-dark:#E1E4E8;">: {</span></span>
<span class="line"><span style="--shiki-light:#005CC5;--shiki-dark:#79B8FF;">            &quot;in_genomes&quot;</span><span style="--shiki-light:#24292E;--shiki-dark:#E1E4E8;">: </span><span style="--shiki-light:#005CC5;--shiki-dark:#79B8FF;">48</span><span style="--shiki-light:#24292E;--shiki-dark:#E1E4E8;">,</span></span>
<span class="line"><span style="--shiki-light:#005CC5;--shiki-dark:#79B8FF;">            &quot;gene_names&quot;</span><span style="--shiki-light:#24292E;--shiki-dark:#E1E4E8;">: [</span><span style="--shiki-light:#032F62;--shiki-dark:#9ECBFF;">&quot;GapA&quot;</span><span style="--shiki-light:#24292E;--shiki-dark:#E1E4E8;">, </span><span style="--shiki-light:#032F62;--shiki-dark:#9ECBFF;">&quot;dnaX&quot;</span><span style="--shiki-light:#24292E;--shiki-dark:#E1E4E8;">],</span></span>
<span class="line"><span style="--shiki-light:#005CC5;--shiki-dark:#79B8FF;">            &quot;classification&quot;</span><span style="--shiki-light:#24292E;--shiki-dark:#E1E4E8;">: </span><span style="--shiki-light:#032F62;--shiki-dark:#9ECBFF;">&quot;single copy core&quot;</span><span style="--shiki-light:#24292E;--shiki-dark:#E1E4E8;">,</span></span>
<span class="line"><span style="--shiki-light:#005CC5;--shiki-dark:#79B8FF;">            &quot;var_sites&quot;</span><span style="--shiki-light:#24292E;--shiki-dark:#E1E4E8;">: </span><span style="--shiki-light:#005CC5;--shiki-dark:#79B8FF;">true</span><span style="--shiki-light:#24292E;--shiki-dark:#E1E4E8;">,</span></span>
<span class="line"><span style="--shiki-light:#005CC5;--shiki-dark:#79B8FF;">            &quot;inf_sites&quot;</span><span style="--shiki-light:#24292E;--shiki-dark:#E1E4E8;">: </span><span style="--shiki-light:#005CC5;--shiki-dark:#79B8FF;">false</span></span>
<span class="line"><span style="--shiki-light:#24292E;--shiki-dark:#E1E4E8;">        }</span></span>
<span class="line"><span style="--shiki-light:#24292E;--shiki-dark:#E1E4E8;">    }</span></span>
<span class="line"><span style="--shiki-light:#24292E;--shiki-dark:#E1E4E8;">]</span></span></code></pre></div><p>Important: When using an array of strings as metadata, be sure to consistently use an array across all homologies.</p><h3 id="tree-files-optional" tabindex="-1"><code>tree files</code> (OPTIONAL) <a class="header-anchor" href="#tree-files-optional" aria-label="Permalink to &quot;`tree files` (OPTIONAL)&quot;">​</a></h3><p>The PanVA frontend can be configured to display one or more additional trees, such as CoreSNP. The files are in Newick format and should be placed in the root of the app directory.</p><h2 id="homology-group-files" tabindex="-1">Homology group files <a class="header-anchor" href="#homology-group-files" aria-label="Permalink to &quot;Homology group files&quot;">​</a></h2><h4 id="alignments-csv" tabindex="-1"><code>alignments.csv</code> <a class="header-anchor" href="#alignments-csv" aria-label="Permalink to &quot;`alignments.csv`&quot;">​</a></h4><p>This is a matrix of the aligned gene sequences and position specific attributes. For example:</p><table><thead><tr><th><code>mRNA_id</code></th><th><code>genome_nr</code></th><th><code>position</code></th><th><code>nucleotide</code></th></tr></thead><tbody><tr><td>97_1_FEDMPDKE_03607_mRNA</td><td>97</td><td>1</td><td>A</td></tr><tr><td>97_1_FEDMPDKE_03607_mRNA</td><td>97</td><td>2</td><td>T</td></tr></tbody></table><ul><li><code>mRNA_id</code>: A unique identifier for each sequence in the homology group (<em>string</em>).</li><li><code>genome_nr</code>: A unique ID for each genome sequence (<em>integer</em>).</li><li><code>position</code>: The position in the alignment (<em>integer</em>).</li><li><code>nucleotide</code>: The nucleotide value (<em>string containing A, C, G, T, or -</em>).</li></ul><p>This file can be extended with <a href="#additionalmetadata">additional metadata</a>.</p><h3 id="sequences-csv" tabindex="-1"><code>sequences.csv</code> <a class="header-anchor" href="#sequences-csv" aria-label="Permalink to &quot;`sequences.csv`&quot;">​</a></h3><p>The sequences extracted from the multiple sequence alignment. This file is used to generate dendrograms. For example:</p><table><thead><tr><th><code>mRNA_id</code></th><th><code>nuc_trimmed_seq</code></th></tr></thead><tbody><tr><td>97_1_FEDMPDKE_03607_mRNA</td><td>ATGAGTTTTGATAATTCCCCACAATCACGCCTGATCCTAACCATGATGGGAGCC...</td></tr><tr><td>87_1_JABOGBIO_03490_mRNA</td><td>ATGAGTTTTGATAATTCCCCACAATCACGCCTGATCCTAACCATGATGGGAGCC...</td></tr></tbody></table><ul><li><code>mRNA_id</code>: A unique identifier for each sequence in the homology group (<em>string</em>).</li><li><code>nuc_trimmed_seq</code>: The nucleotide sequences, trimmed in PanTools (<em>string</em>).</li></ul><h3 id="variable-csv" tabindex="-1"><code>variable.csv</code> <a class="header-anchor" href="#variable-csv" aria-label="Permalink to &quot;`variable.csv`&quot;">​</a></h3><p>Summary of all variable positions in the alignment and their value counts. This data is used for calculation of the conservation score at each aligned position. For example:</p><table><thead><tr><th><code>position</code></th><th><code>informative</code></th><th><code>A</code></th><th><code>C</code></th><th><code>G</code></th><th><code>T</code></th><th><code>gap</code></th><th><code>pheno_specific</code></th></tr></thead><tbody><tr><td>1</td><td>True</td><td>0</td><td>30</td><td>70</td><td>0</td><td>0</td><td>True</td></tr><tr><td>9</td><td>False</td><td>1</td><td>0</td><td>70</td><td>99</td><td>0</td><td>False</td></tr></tbody></table><ul><li><code>position</code>: The position in the alignment (<em>integer</em>).</li><li><code>informative</code>: Is position informative in nucleotide alignment (<em>boolean</em>).</li><li><code>A</code>: Number of sequences containing nucleotide A (<em>integer</em>).</li><li><code>C</code>: Number of sequences containing nucleotide C (<em>integer</em>).</li><li><code>G</code>: Number of sequences containing nucleotide G (<em>integer</em>).</li><li><code>T</code>: Number of sequences containing nucleotide T (<em>integer</em>).</li><li><code>gap</code>: Number of sequences containing a gap (-) (<em>integer</em>).</li></ul><p>This file can be extended with <a href="#additionalmetadata">additional metadata</a>.</p><h3 id="annotations-csv-optional" tabindex="-1"><code>annotations.csv</code> (OPTIONAL) <a class="header-anchor" href="#annotations-csv-optional" aria-label="Permalink to &quot;`annotations.csv` (OPTIONAL)&quot;">​</a></h3><p>This optional file is only used for Eukaryotic pangenomes. It specifies the gene models matched to each gene sequences of reference genomes (for which GFF files are available). For example:</p><table><thead><tr><th><code>mRNA_id</code></th><th><code>position</code></th><th><code>cds</code></th><th><code>exon</code></th></tr></thead><tbody><tr><td>5_1_ATERI-1G45130.1</td><td>1</td><td>True</td><td>False</td></tr><tr><td>5_1_ATERI-1G45130.2</td><td>1</td><td>False</td><td>False</td></tr></tbody></table><ul><li><code>mRNA_id</code>: A unique identifier for each sequence in the homology group (<em>string</em>).</li><li><code>position</code>: The position in the alignment (<em>integer</em>).</li></ul><p>This file can be extended with <a href="#additionalmetadata">additional metadata</a>.</p><p>This file can be extended with additional columns to be used as annotations. For example:</p><ul><li><code>cds</code>: Does position have this feature in nucleotide alignment (<em>boolean</em>).</li><li><code>exon</code>: Does position have this feature in nucleotide alignment (<em>boolean</em>).</li></ul><p>The frontend needs to be <a href="./../../frontend/docs/config.html">configured</a> to display these annotations.</p><h3 id="metadata-csv-optional" tabindex="-1"><code>metadata.csv</code> (OPTIONAL) <a class="header-anchor" href="#metadata-csv-optional" aria-label="Permalink to &quot;`metadata.csv` (OPTIONAL)&quot;">​</a></h3><p>An optional CSV file containing metadata for each genome indicated by <code>mRNA_id</code> that should be included in the analysis. For example:</p><table><thead><tr><th><code>mRNA_id</code></th><th><code>virulence</code></th><th><code>species</code></th></tr></thead><tbody><tr><td>97_1_FEDMPDKE_03607_mRNA</td><td>avirulent</td><td>P.brasiliens</td></tr><tr><td>87_1_JABOGBIO_03490_mRNA</td><td>?</td><td>P.brasiliense</td></tr></tbody></table><ul><li><code>mRNA_id</code>: A unique identifier for each sequence in the homology group (<em>string</em>).</li></ul><p>This file can be extended with <a href="#additionalmetadata">additional metadata</a>.</p><h3 id="linkage-matrix-npy-auto-generated" tabindex="-1"><code>linkage_matrix.npy</code> (AUTO-GENERATED) <a class="header-anchor" href="#linkage-matrix-npy-auto-generated" aria-label="Permalink to &quot;`linkage_matrix.npy` (AUTO-GENERATED)&quot;">​</a></h3><p>The linkage matrix for generating the initial clustering dendrogram, stored as NumPy file. This file is generated once by the API and is used to improve application performance.</p><p><strong>Important:</strong> Please delete this file if the contents of <code>sequences.csv</code> has changed.</p><h2 id="additional-metadata" tabindex="-1">Additional metadata <a class="header-anchor" href="#additional-metadata" aria-label="Permalink to &quot;Additional metadata&quot;">​</a></h2><p>Some files, specifically <a href="#alignmentscsv"><code>alignments.csv</code></a>, <a href="#metadatacsv"><code>metadata.csv</code></a>, and <a href="#variablecsv"><code>variable.csv</code></a>, can be extended with additional metadata columns. Values in each column should be of the same type (string, number, optional boolean).</p><p>The frontend needs to be <a href="./../../frontend/docs/config.html">configured</a> to use these additional columns.</p></div></div></main><footer class="VPDocFooter" data-v-b474ad16 data-v-26f873fc><!--[--><!--]--><div class="edit-info" data-v-26f873fc><!----><div class="last-updated" data-v-26f873fc><p class="VPLastUpdated" data-v-26f873fc data-v-a0cec9fc>Last updated: <time datetime="2024-04-23T21:26:37.000Z" data-v-a0cec9fc></time></p></div></div><nav class="prev-next" data-v-26f873fc><div class="pager" data-v-26f873fc><a class="VPLink link pager-link prev" href="/PanVA/v0.0.0/data-format.html" data-v-26f873fc><!--[--><span class="desc" data-v-26f873fc>Previous page</span><span class="title" data-v-26f873fc>Data Format Reference</span><!--]--></a></div><div class="pager" data-v-26f873fc><a class="VPLink link pager-link next" href="/PanVA/v0.0.0/config.html" data-v-26f873fc><!--[--><span class="desc" data-v-26f873fc>Next page</span><span class="title" data-v-26f873fc>Configuration</span><!--]--></a></div></nav></footer><!--[--><!--]--></div></div></div><!--[--><!--]--></div></div><footer class="VPFooter has-sidebar" data-v-014bb8c7 data-v-020d9df9><div class="container" data-v-020d9df9><p class="message" data-v-020d9df9>Released under the XXX License. Docs built with <a href="https://vitepress.dev">VitePress</a>.</p><p class="copyright" data-v-020d9df9>Copyright © 2024-present, PanVA team, Wageningen University & Eindhoven University of Technology.</p></div></footer><!--[--><!--]--></div></div>
    <script>window.__VP_HASH_MAP__=JSON.parse("{\"v0.0.0_data-format-homology.md\":\"D4-vdtjc\",\"v0.0.0_markdown-examples.md\":\"Dze1dY6q\",\"v0.0.0_api-examples.md\":\"mo0diWja\",\"v0.0.0_install.md\":\"D2GP1805\",\"v0.0.0_config.md\":\"B7YJPNOs\",\"demos.md\":\"Btx2-tHI\",\"v0.0.0_data-format-geneset.md\":\"Bn1ViaZj\",\"v0.0.0_data-format.md\":\"C59IoLMC\",\"v0.0.0_dev-setup.md\":\"DM-Lkt0j\",\"index.md\":\"ecEojTCv\"}");window.__VP_SITE_DATA__=JSON.parse("{\"lang\":\"en-US\",\"dir\":\"ltr\",\"title\":\"PanVA\",\"description\":\"Documentation for PanVA\",\"base\":\"/PanVA/\",\"head\":[],\"router\":{\"prefetchLinks\":true},\"appearance\":true,\"themeConfig\":{\"search\":{\"provider\":\"local\"},\"nav\":[{\"text\":\"Home\",\"link\":\"/\"},{\"text\":\"Installation\",\"link\":\"/v0.0.0/install\"},{\"text\":\"Demos\",\"link\":\"/demos\"},{\"text\":\"Tutorials\",\"link\":\"/v0.0.0/tutorials\"},{\"text\":\"Versions\",\"items\":[{\"text\":\"0.0.0 (latest)\",\"link\":\"/v0.0.0/install\"}]}],\"sidebar\":{\"/v0.0.0/\":[{\"text\":\"Getting Started\",\"collapsed\":false,\"items\":[{\"text\":\"Installation\",\"link\":\"/v0.0.0/install\"}]},{\"text\":\"User Guide\",\"collapsed\":false,\"items\":[{\"text\":\"Data Format Reference\",\"link\":\"/v0.0.0/data-format\",\"items\":[{\"text\":\"Homology\",\"link\":\"/v0.0.0/data-format-homology\"}]},{\"text\":\"Configuration\",\"link\":\"/v0.0.0/config\",\"items\":[{\"text\":\"Example config\",\"link\":\"/v0.0.0/config#example-config\"}]},{\"text\":\"Server setup\",\"link\":\"/v0.0.0/server-config\"}]},{\"text\":\"Demos & Tutorials\",\"collapsed\":false,\"items\":[{\"text\":\"Yeast\",\"link\":\"/v0.0.0/yeast-tutorial\"},{\"text\":\"Pectobacterium\",\"link\":\"/v0.0.0/pecto-tutorial\"},{\"text\":\"Arabidopsis\",\"link\":\"/v0.0.0/ara-tutorial\"},{\"text\":\"PanTools\",\"link\":\"/pantools\"}]},{\"text\":\"Development Guide\",\"link\":\"v0.0.0/dev-setup\"}],\"/v0.0.1/\":[{\"text\":\"Getting Started\",\"collapsed\":false,\"items\":[{\"text\":\"Installation\",\"link\":\"/installation\"},{\"text\":\"...\",\"link\":\"/api-examples\"}]},{\"text\":\"User Guide\",\"collapsed\":false,\"items\":[{\"text\":\"Data Format Reference\",\"link\":\"/data-format\",\"items\":[{\"text\":\"Data Format Reference\",\"link\":\"/data-format-homology\"}]}]},{\"text\":\"Examples\",\"collapsed\":false,\"items\":[{\"text\":\"Configuration\",\"link\":\"/config\"},{\"text\":\"Server setup\",\"link\":\"/server-config\"},{\"text\":\"Demos\",\"link\":\"/demos\"}]},{\"text\":\"Tutorials\",\"collapsed\":false,\"items\":[{\"text\":\"Yeast\",\"link\":\"/yeast-tutorial\"},{\"text\":\"Pectobacterium\",\"link\":\"/pecto-tutorial\"},{\"text\":\"Arabidopsis\",\"link\":\"/ara-tutorial\"},{\"text\":\"PanTools\",\"link\":\"/pantools\"}]}]},\"socialLinks\":[{\"icon\":\"github\",\"link\":\"https://github.com/PanBrowse/PanVA\"}],\"footer\":{\"message\":\"Released under the XXX License. Docs built with <a href=\\\"https://vitepress.dev\\\">VitePress</a>.\",\"copyright\":\"Copyright © 2024-present, PanVA team, Wageningen University & Eindhoven University of Technology.\"}},\"locales\":{},\"scrollOffset\":134,\"cleanUrls\":false}");</script>
    
  </body>
</html>